internal/mode/static/telemetry/job_test.go

```diff
@@ -42,20 +68,42 @@ var _ = Describe("Job", func() {
 				errCh <- job.Start(ctx)
 				close(errCh)
 			}()
+			time.Sleep(sleep)
+			close(readyChannel)
 
 			const minReports = 2 // ensure that the Job reports more than once: it doesn't exit after the first report
 
 			Eventually(exporter.ExportCallCount).Should(BeNumerically(">=", minReports))
 			for i := 0; i < minReports; i++ {
 				_, data := exporter.ExportArgsForCall(i)
-				Expect(data).To(Equal(telemetry.Data{}))
+				Expect(data).To(Equal(expData))
 			}
 
 			cancel()
 			Eventually(errCh).Should(Receive(BeNil()))
 			Eventually(errCh).Should(BeClosed())
 		},
-		Entry("Job runs with Exporter not returning errors", nil),
-		Entry("Job runs with Exporter returning an error", errors.New("some error")),
+		Entry("Job runs with Exporter not returning errors", nil, time.Duration(0)),
+		Entry("Job runs with Exporter returning an error", errors.New("some error"), time.Duration(0)),
+		Entry("Job runs with extended time waiting for NGF Pod to be ready", nil, 500*time.Millisecond),
 	)
+
+	When("job context is canceled", func() {
+		It("should gracefully exist", func() {
+			ctx, cancel := context.WithTimeout(context.Background(), timeout)
+
+			errCh := make(chan error)
+			go func() {
+				errCh <- job.Start(ctx)
+				close(errCh)
+			}()
+
+			sleep := 500 * time.Millisecond
+			time.Sleep(sleep)
+			cancel()
+
+			Eventually(errCh).Should(Receive())
```

Comment:
Let's check that the error is of type `context.Canceled` here.

Reply:
no longer relevant as tests have changed.